File: old/v1.0/java/Screen.java
public class Screen implements Asciigame
{
	public static final Point MESSAGE = new Point(25,0);

	public static void clear()
	{
		System.out.print("\033[1;1H\033[2J");
	}

	public static void clearLine(Point locat)
	{
		Screen.set(locat);
		System.out.print("\033[2K");
	}

	public static void saveAndClear()
	{
		System.out.print("\033[?47h");
	}

	public static void restore()
	{
		System.out.print("\033[?47l");
	}

	public static void set(Point locat)
	{
		System.out.printf("\033[%d;%dH",locat.v,locat.h);
	}

	public static void printat(Point locat, char c)
	{
		Screen.set(locat);
		System.out.write(c);
	}
}

File: old/v1.0/c/screen.c
#include <stdio.h>
#include "asciigame.h"

void screen_set(struct point locat)
{
	printf("\033[%d;%dH",locat.v,locat.h);
}

void clear_screen(void)
{
	printf("\e[1;1H\e[2J");
}

void clear(struct point locat, int x)
{
	screen_set(locat);
	printf("\e[%dK",x);
}

int printat(char *str, struct point locat)
{
	screen_set(locat);
	return printf("%s",str);
}

int putcharat(char s, struct point locat)
{
	screen_set(locat);
	return putchar(s);
}
